craft_grammar/_processor.py

```diff
@@ -253,12 +253,21 @@ def _parse_section_dictionary(
                 # markers to indicate the start or change of statement,
                 # the current statement is complete and this section
                 # is a primitive to be collected.
-                finalized_statement = statement
+                finalized_statement = _finalize(statement, finalized_statement)
                 statement = None
 
         return statement, finalized_statement
 
 
+def _finalize(
```

Comment:
The change looks conceptually fine to me, but I worry about unintended breakages. Can you run Snapcraft's tests with this branch?

Reply:
I ran Snapcraft's tests with the branch and found no failure related to grammar.